tests/core/test_accounts.py

```diff
@@ -481,23 +507,30 @@ def test_eth_account_encrypt(
 
 
 @pytest.mark.parametrize(
-    'private_key, password, kdf, expected_decrypted_key, expected_kdf',
-    encrypt_mark_params(),
-    ids=['hex_str', 'hex_str_provided_kdf', 'eth_keys.datatypes.PrivateKey']
+    'private_key, password, kdf, iterations, expected_decrypted_key, expected_kdf',
+    get_encrypt_test_params(),
+    ids=[
+        'hex_str',
+        'hex_str_provided_kdf',
+        'eth_keys.datatypes.PrivateKey',
+        'hex_str_pbkdf2_provided_iterations',
+        'hex_str_scrypt_provided_iterations',
+    ]
 )
 def test_eth_account_prepared_encrypt(
         acct,
         private_key,
         password,
         kdf,
+        iterations,
         expected_decrypted_key,
         expected_kdf):
     account = acct.privateKeyToAccount(private_key)
 
     if kdf is None:
-        encrypted = account.encrypt(password)
+        encrypted = account.encrypt(password, iterations=iterations)
     else:
-        encrypted = account.encrypt(password, kdf)
+        encrypted = account.encrypt(password, kdf=kdf, iterations=iterations)
 
     assert encrypted['address'] == '2c7536e3605d9c16a7a3d7b1898e529396a65c23'
     assert encrypted['version'] == 3
```

Comment:
Just after the kdf assertion looks like the right place to verify the iterations in the encrypted keyfile json.

```
In [3]: create_keyfile_json(b'1'*32, b'aonteuhsantoheu', kdf='scrypt', iterations=128)
Out[3]: 
{'address': '77952ce83ca3cad9f7adcfabeda85bd2f1f52008',
 'crypto': {'cipher': 'aes-128-ctr',
  'cipherparams': {'iv': 'b38a0dc532c258ae55a08b1967a03e23'},
  'ciphertext': 'cbebc97f6ef6642fbc9e77bce055d41980ddae28ddf3e1cca95402035ccce94b',
  'kdf': 'scrypt',
  'kdfparams': {'dklen': 32,
   'n': 128,  # <---------------------------------------------------- iterations shows up here
   'p': 8,
   'r': 1,
   'salt': '27f214a3a0ca8b07f3168902e449417d'},
  'mac': 'bf22caaa85025c22b9081326f4e7646cb78ec369540ce31fe2c3130d67274a75'},
 'id': 'c7d68f4a-0b4b-460f-b128-6ef5b03816c1',
 'version': 3}

In [4]: create_keyfile_json(b'1'*32, b'aonteuhsantoheu', iterations=128)
Out[4]: 
{'address': '77952ce83ca3cad9f7adcfabeda85bd2f1f52008',
 'crypto': {'cipher': 'aes-128-ctr',
  'cipherparams': {'iv': '8fe849a2d481c224493fdb4e0284cfd5'},
  'ciphertext': 'b1f45f5ff9a81479237a241ccdc9efde073cb663d3461a06f2cd7ea0dfc487a0',
  'kdf': 'pbkdf2',
  'kdfparams': {'c': 128,  # <---------------------------------------- iterations shows up here
   'dklen': 32,
   'prf': 'hmac-sha256',
   'salt': '70dd2aa94c18583833b558814f1ce3fa'},
  'mac': '1d5b5d1977da7702e4592a0d8ce3c47da1fcb0ab15b055ac37e346c0a3bc011c'},
 'id': '0b9afb25-e29d-4ed6-ba51-8dfbb98c3362',
 'version': 3}

Reply:
Thanks for pointing that out! See comment here: https://github.com/ethereum/eth-account/pull/52/commits/aa9182811a67a673dc428f1f37ea41fad14754fc#r247003981